src/app/static/src/app/store/modelOptions/modelOptions.ts

```diff
@@ -4,20 +4,23 @@ import {DynamicFormData, DynamicFormMeta} from "../../components/forms/types";
 import {mutations} from "./mutations";
 import {localStorageManager} from "../../localStorageManager";
 import {actions} from "./actions";
+import {VersionInfo} from "../../generated";
 
 export interface ModelOptionsState {
     optionsFormMeta: DynamicFormMeta
     options: DynamicFormData
     valid: boolean
     fetching: boolean
+    version: VersionInfo
 }
 
 export const initialModelOptionsState = (): ModelOptionsState => {
     return {
         optionsFormMeta: {controlSections: []},
         options: {},
         valid: false,
-        fetching: false
+        fetching: false,
+        version: {hintr: "unknown", naomi: "unknown", rrq: "unknown"}
```

Comment:
If we're expecting particular property names in VersionInfo in the front end, should there be a correspondingly strong type in the back end, rather than just Map<String, String>? I guess the backend doesn't care much what the properties are and will just let hintr error if the format is wrong.

Reply:
Well the front-end is the bit that is actually hooked up to the schema - i.e. we don't have a way of auto-generating kotlin models from the schema. So this way is more flexible - if the schema changes, just have to regenerate the types and the backend will still work.